src/borg/testsuite/archiver.py

```diff
@@ -1528,17 +1529,44 @@ def test_corrupted_repository(self):
         output = self.cmd('check', '--info', self.repository_location, exit_code=1)
         self.assert_in('Starting repository check', output)  # --info given for root logger
 
-    # we currently need to be able to create a lock directory inside the repo:
-    @pytest.mark.xfail(reason="we need to be able to create the lock directory inside the repo")
-    def test_readonly_repository(self):
+    @unittest.skipUnless(sys.platform.startswith('linux'), 'readonly tests require linux')
+    def test_readonly_list(self):
         self.cmd('init', '--encryption=repokey', self.repository_location)
         self.create_src_archive('test')
-        os.system('chmod -R ugo-w ' + self.repository_path)
-        try:
-            self.cmd('extract', '--dry-run', self.repository_location + '::test')
-        finally:
-            # Restore permissions so shutil.rmtree is able to delete it
-            os.system('chmod -R u+w ' + self.repository_path)
+        with self.read_only(self.repository_path):
+            # verify that list normally doesn't work with read-only repo
+            if self.FORK_DEFAULT:
+                self.cmd('list', self.repository_location, exit_code=EXIT_ERROR)
+            else:
+                with pytest.raises((LockFailed, RemoteRepository.RPCError)) as excinfo:
+                    self.cmd('list', self.repository_location)
+                if isinstance(excinfo.value, RemoteRepository.RPCError):
+                    assert excinfo.value.exception_class == 'LockFailed'
+            # verify that list works with read-only repo when using --bypass-lock
+            self.cmd('list', self.repository_location, '--bypass-lock')
+
+    @unittest.skipUnless(sys.platform.startswith('linux'), 'readonly tests require linux')
+    @unittest.skipUnless(has_llfuse, 'llfuse not installed')
+    def test_readonly_mount(self):
+        self.cmd('init', '--encryption=repokey', self.repository_location)
+        self.create_src_archive('test')
+        with self.read_only(self.repository_path):
+            mountpoint = os.path.join(self.tmpdir, 'mountpoint')
+            # verify that mount normally doesn't work with read-only repo
+            try:
+                os.mkdir(mountpoint)
+                if self.FORK_DEFAULT:
+                    self.cmd('mount', self.repository_location, mountpoint, exit_code=EXIT_ERROR)
+                else:
+                    with pytest.raises((LockFailed, RemoteRepository.RPCError)) as excinfo:
+                        self.cmd('mount', self.repository_location, mountpoint)
+                    if isinstance(excinfo.value, RemoteRepository.RPCError):
+                        assert excinfo.value.exception_class == 'LockFailed'
+            finally:
+                os.rmdir(mountpoint)
+            # verify that mount works with read-only repo when using --bypass-lock
+            with self.fuse_mount(self.repository_location, mountpoint, '--bypass-lock'):
+                pass
```

Comment:
this is somehow different way than in the block above. also, block above kills mountpoint in finally block.

Reply:
`self.fuse_mount` is the "complete" test including mkdir, mount, umount, rmdir. (Reusing that functiong from other tests.) The block above already fails at mount, so that's what is explicitly being tested. Do you want me to rewrite `fuse_mount` so it works with testing a failed mount?

Reply:
i just wondered why it is not calling via self.cmd.

Reply:
It is calling via `self.cmd` but by using `self.fuse_mount` ;) So that is resolved? Or you do suggest changes to clarify?

Reply:
> i just wondered why it is not calling via self.cmd.

Reflecting on that, I think you're right. It's unnecessarily complex to use different calls. I have modified `self.fuse_mount` to make it accept keyword arguments which are then being passed on to `self.cmd`. This works and make the `test_readonly_mount` look cleaner. :+1: